src/graphnet/data/utilities/parquet_to_sqlite.py

```diff
@@ -0,0 +1,227 @@
+        self._save_config(outdir, database_name)
+        print(
+            f"Database saved at \n{outdir}/{database_name}/data/{database_name}.db"
+        )
+
+    def _count_events(self, open_parquet_file: str = None):
+        return len(open_parquet_file[self._mc_truth_table])
+
+    def _save_to_sql(
+        self,
+        outdir: str = None,
+        database_name: str = None,
+        ak_array: awkward.Array = None,
+        field_name: str = None,
+        n_events_in_file: int = None,
+    ):
+        df = self._make_df(ak_array, field_name, n_events_in_file)
+        if field_name in self._created_tables:
+            self._submit_to_database(outdir, database_name, field_name, df)
+        else:
+            self._create_table(
+                df, field_name, outdir, database_name, n_events_in_file
+            )
+            self._created_tables.append(field_name)
+            self._submit_to_database(outdir, database_name, field_name, df)
+
+    def _make_df(
+        self,
+        ak_array: awkward.Array = None,
+        field_name: str = None,
+        n_events_in_file: int = None,
+    ):
+        df = pd.DataFrame(ak.to_pandas(ak_array[field_name]))
+        if len(df.columns) == 1:
+            if df.columns == ["values"]:
+                df.columns = [field_name]
+        if (
+            len(df) != n_events_in_file
+        ):  # if true, the dataframe contains more than 1 row pr. event (e.g. Pulsemap).
+            event_nos = []
+            for event_no in range(n_events_in_file + self._event_counter):
+                try:
+                    event_nos.extend(
+                        np.repeat(
+                            event_no, len(df[df.columns[0]][event_no])
+                        ).tolist()
+                    )
+                except KeyError:  # KeyError indicates that this df has no entry for event_no (e.g. an event with no detector response)
+                    pass
+        else:
+            event_nos = np.arange(0, n_events_in_file, 1) + self._event_counter
+        df["event_no"] = event_nos
+        return df
+
+    def _submit_to_database(
+        self, outdir: str, database_name: str, key: str, data: pd.DataFrame
+    ):
+        """Submits data to the database with specified key."""
+        if len(data) == 0:
+            if self._verbose:
+                self.logger.info(f"No data provided for {key}.")
+            return
+        engine = sqlalchemy.create_engine(
+            "sqlite:///"
+            + outdir
+            + "/"
+            + database_name
+            + "/data/"
+            + database_name
+            + ".db"
+        )
+        data.to_sql(key, engine, index=False, if_exists="append")
+        engine.dispose()
+
+    def _create_table(
+        self,
+        df: pd.DataFrame,
+        field_name: str = None,
+        outdir: str = None,
+        database_name: str = None,
+        n_events_in_file: int = None,
+    ):
+        """Creates a table.
```

Comment:
This appears duplicate of https://github.com/graphnet-team/graphnet/blob/main/src/graphnet/data/sqlite/sqlite_dataconverter.py#L120-L146. Since `self` isn't being referenced, I figure we could convert this into a utility function in `graphnet.data.sqlite.sqlite_utilities` and use that method both here and in `SQLiteDataConverter`,